units/stats_pod.go

```diff
@@ -0,0 +1,104 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/pod"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const podStatsCollectorJobName = "pod-stats-collector"
+
+func init() {
+	registry.AddJobType(podStatsCollectorJobName, func() amboy.Job {
+		return makePodStatsCollector()
+	})
+}
+
+type podStatsCollector struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+}
+
+// NewPodStatsCollector logs statistics about current pod usage.
+func NewPodStatsCollector(id string) amboy.Job {
+	j := makePodStatsCollector()
+	j.SetID(fmt.Sprintf("%s.%s", podStatsCollectorJobName, id))
+	j.SetEnqueueScopes(podStatsCollectorJobName)
+
+	return j
+}
+
+func makePodStatsCollector() *podStatsCollector {
+	j := &podStatsCollector{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    podStatsCollectorJobName,
+				Version: 0,
+			},
+		},
+	}
+	return j
+}
+
+func (j *podStatsCollector) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		j.AddError(err)
```

Comment:
Can we instead use `errors.Wrap(err, "getting service flags")`?

Reply:
Done.